Reflects API allowances for the organization in BillingDetails

gib opened the pull request:
First step to addressing https://github.com/aptible/dashboard.aptible.com/issues/449. Once this is in, a dashboard update will be required.

@sandersonet - I added some more logic for the `pilot` plan. Specifically to `allowPHI`. Let me know if I went too far on this one and also the development plan check -- do we offer any allowances for development plans? I assumed no (see comment in the diff).

gib committed:
Reflects API allowances for the organization in BillingDetails

First step to addressing
https://github.com/aptible/dashboard.aptible.com/issues/449. Once this
is in, a dashboard update will be required.

@sandersonet - I added some more logic for the `pilot` plan.
Specifically to `allowPHI`. Let me know if I went too far on this one.

gib commented on app/models/billing-detail.js:
```diff
@@ -46,7 +49,7 @@ export default DS.Model.extend({
   billingContact: DS.belongsTo('user', {async:true}),
   organization: DS.belongsTo('organization', {async:true}),
-  allowPHI: Ember.computed.match('plan', /production|platform/),
+  allowPHI: Ember.computed.match('plan', /pilot|production|platform/),
```
:+1: or :-1: ?

skylar-anderson replied:
:+1:

gib commented on app/models/billing-detail.js:
```diff
@@ -59,13 +62,28 @@ export default DS.Model.extend({
+    if (plan === 'development' || !CONTAINER_ALLOWANCES.hasOwnProperty(plan)) {
+      return 0;
+    }
+    return this.get('containerAllowance') || CONTAINER_ALLOWANCES[plan];
```
Is the `development` assumption correct here or are there cases where an account could have development resource allowances?

skylar-anderson replied:
No, development will always be 0.

I don't think you need to check that `CONTAINER_ALLOWANCES` has the plan key before using it.  Instead maybe just return `this.get('containerAllowance') || CONTAINER_ALLOWANCES[plan] || 0;`

skylar-anderson replied:
Sorry, I looked at the tests and realize what you are implying here.  I think we should always look to the value of what's in the API first, regardless of plan.  So, it _could_ be possible that we custom configure a development plan with allowances.

gib committed:
Cleaning up BillingDetail updates

Thank you @sandersonet, applying your feedback here:
- Assume development will always have 0 allowance values.
- No longer explicitly check for plan type existence.
- Catch all to return 0 allowances in cases of undefined values

gib commented:
I'm going to go ahead and merge this.

skylar-anderson commented:
:+1: Great tests
